baseapp/baseapp.go

```diff
@@ -593,6 +597,36 @@ func (app *BaseApp) validateFinalizeBlockHeight(req *abci.RequestFinalizeBlock)
 	return nil
 }
 
+// validate that the tx is not signed using SIGN_MODE_AMINO_JSON if the tx is nonAtomic
+func (app *BaseApp) validateNonAtomicSignMode(tx sdk.Tx) error {
```

Comment:
nit: this method doesn't rely on `BaseApp` as far as I can tell. So,

1. Consider moving this elsewhere? maybe `x/tx`?
2. If it must stay in baseapp, then consider making this a function

Reply:
left it here for now since its provate, but in the future with runtime it will move